### PR TITLE
Send telemetry/event to client first time Qute template is opened.

### PR DESCRIPTION
**Pull request by rgrunber**

### CODE REVIEW

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TelemetryEvent.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class TelemetryEvent {
+    private String name;
+    private Object properties;
+
+    public TelemetryEvent(String name, Object properties) {
+        this.name = name;
+        this.properties = properties;
+    }
+
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TelemetryEvent.java:22]: Please add javadoc which explain what is the name

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TelemetryEvent.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public String getName() {
+        return name;
+    }
+
+    public Object getProperties() {
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TelemetryEvent.java:27]: Please add javadoc which explain what is the properties

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java [hunk 3]
```diff
@@ -100,6 +102,10 @@ public TemplateFileTextDocumentService(QuteLanguageServer quteLanguageServer, Sh
 	@Override
 	public void didOpen(DidOpenTextDocumentParams params) {
 		QuteOpenedTextDocument document = (QuteOpenedTextDocument) documents.onDidOpenTextDocument(params);
+		if (!hasOpenedAQuteDocument) {
+			hasOpenedAQuteDocument = true;
+			getLanguageClient().telemetryEvent(new TelemetryEvent("qute.file.opened", new HashMap<>()));
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java:107]: declare this string in a new class like QuteTelementryConstants

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java [hunk 4]
```diff
@@ -100,6 +104,10 @@ public TemplateFileTextDocumentService(QuteLanguageServer quteLanguageServer, Sh
 	@Override
 	public void didOpen(DidOpenTextDocumentParams params) {
 		QuteOpenedTextDocument document = (QuteOpenedTextDocument) documents.onDidOpenTextDocument(params);
+		if (!hasOpenedAQuteDocument) {
+			hasOpenedAQuteDocument = true;
+			getLanguageClient().telemetryEvent(new TelemetryEvent(QuteTelemetryConstants.FILE_OPENED, new HashMap<>()));
```
Comment (datho7561) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java:109]: `QuteTelemetryConstants` seems to be missing.

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java (hunk 4 above)
Comment (rgrunber) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java:109]: Fixed.

**Review by datho7561 — APPROVED** [review]
Looks good, and it works well. Thanks, Roland!

**Pull request merged**